### PR TITLE
HPCC-16173 Add tracing to note lost UDP packets in roxie log

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@fvillanustre FYI
@mckellyln Please review

File: system/jlib/jdebug.cpp [hunk 1]
```diff
@@ -2019,6 +2019,169 @@ static struct CNtKernelInformation
...
+                else if (streq(columnNames.item(idx2), "local_port"))
+                    portCol = idx2;
+                else if (streq(columnNames.item(idx2), "rx_queue"))
+                    queueCol = idx2;
+                else if (streq(columnNames.item(idx2), "uid"))
+                    uidCol = idx2;
+            }
+            if (portCol == -1 || queueCol == -1 || uidCol == -1)
+            {
+                portCol = -1;
```
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: portCol or uidCol here ?

File: system/jlib/jdebug.cpp [hunk 2]
```diff
@@ -2019,6 +2019,169 @@ static struct CNtKernelInformation
...
+                    columnNames.add("local_port", idx+1);
+            }
+            ForEachItemIn(idx2, columnNames)
+            {
+                if (streq(columnNames.item(idx2), "drops"))
+                    dropsCol = idx2;
+                else if (streq(columnNames.item(idx2), "local_port"))
+                    portCol = idx2;
+                else if (streq(columnNames.item(idx2), "rx_queue"))
+                    queueCol = idx2;
```
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: Is tx_queue also worth recording ?

File: system/jlib/jdebug.cpp [hunk 3]
```diff
@@ -2019,6 +2019,169 @@ static struct CNtKernelInformation
...
+    CSnmpStatsReporter()
+    {
+        inErrorsCol = -1;
+        prevErrors = 0;
+    }
+    bool reportSnmpInfo()
+    {
+        if (inErrorsCol==-1 && columnNames.length())
+            return false;
+        FILE *netfp = fopen("/proc/net/snmp", "r");
```
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: I wonder if /proc/<roxie-pid>/net/snmp file is same or just for roxie pid

File: system/jlib/jdebug.cpp [hunk 4]
```diff
@@ -2333,6 +2503,11 @@ static class CMemoryUsageReporter: public Thread
                 tm.reset(NAMEDCOUNTPERIOD*1000);
             }
 #endif
+            if (traceMode&PerfMonUDP)
+            {
+                snmpStats.reportSnmpInfo();
+                udpStats.reportUdpInfo(0);
```
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: Should we somehow grab udpTraceLevel setting value from config or roxie and pass in here ?

File: system/jlib/jdebug.cpp [hunk 5]
```diff
@@ -2320,6 +2485,11 @@ static class CMemoryUsageReporter: public Thread
     {
         StringBuffer str;
         getSystemTraceInfo(str, traceMode&~PerfMonExtended); // initializes the values so that first one we print is meaningful rather than always saying PU=0%
+        if (traceMode&PerfMonUDP)
+        {
+            snmpStats.reportSnmpInfo();
+            udpStats.reportUdpInfo(0);
```
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: same as comment below about trace level

**Comment by mckellyln** [comment]
I have one question about line 2077 (uidCol) and some other minor notes.
Otherwise, code looks good, I like your use of appendList.

File: system/jlib/jdebug.cpp [hunk 6]
```diff
@@ -2019,6 +2019,169 @@ static struct CNtKernelInformation
...
+            }
+        }
+        int myUid = geteuid();
+        while (fgets(ln, sizeof(ln), netfp))
+        {
+            StringArray cols;
+            cols.appendList(ln, " :");
+            if (cols.length() >= columnNames.length() && atoi(cols.item(uidCol))==myUid)
+            {
+                unsigned queue = strtoul(cols.item(queueCol), NULL, 16);
```
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: probably not worth checking result of strtoul() call.

File: system/jlib/jdebug.cpp (hunk 2 above)
Comment (richardkchapman) [inline system/jlib/jdebug.cpp]: Probably not

File: system/jlib/jdebug.cpp (hunk 4 above)
Comment (richardkchapman) [inline system/jlib/jdebug.cpp]: No, I want these reports even at minimal tracing

File: system/jlib/jdebug.cpp (hunk 1 above)
Comment (richardkchapman) [inline system/jlib/jdebug.cpp]: Should be uidCol - fixed and repushed

**Comment by richardkchapman** [comment]
I fixed the uidCol issue and repushed.

@ghalliday Please review/merge

**Pull request merged**